File: 389-find-the-difference/find-the-difference.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
public:
    char findTheDifference(string s, string t) {
        unordered_map<char,int> smap;
        unordered_map<char,int> tmap;
        for(int i=0 ; i<t.length(); i++){
            tmap[t[i]]++;
        }
        for(int i=0 ; i<s.length(); i++){
            smap[s[i]]++;
        }
        for(auto x : t){
            if(smap[x]!=tmap[x]  || smap.find(x)==0){
                return x;
            }
        }
        return '\0';
    }
};

File: 389-find-the-difference/find-the-difference.py
class Solution:
    def findTheDifference(self, s: str, t: str) -> str:
        sdict ={}
        tdict ={}
        for x in s:
            if x not in sdict:
                sdict[x]=1
            else:
                sdict[x] = sdict.get(x)+1
        for y in t:
            if y not in tdict:
                tdict[y]=1
            else:
                tdict[y]= tdict.get(y)+1
        for z in t:
            if sdict.get(z)!=tdict.get(z) or z not in sdict:
                return z
            else:
                continue
        return None


        